cfme/tests/infrastructure/test_advanced_search_vms.py

```diff
@@ -333,3 +341,70 @@ def test_cannot_delete_vm_filter_more_than_once(vm_view):
     vm_view.flash.assert_no_error()
     # Try it second time
     assert not vm_view.entities.search.delete_filter(), 'Delete twice accepted!'
+
+
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1506672, 1660460, 1718895])
+def test_create_filter_with_multiple_conditions(appliance, provider, request, running_vms):
+    """
+    Polarion:
+        assignee: anikifor
+        casecomponent: WebUI
+        caseimportance: high
+        initialEstimate: 1/5h
+        setup:
+            1. Navigate to Compute > Infrastructure > Providers.
+            2. Click on Advanced Search Filter.
+        testSteps:
+            1. Create an expression with multiple types of condition. Eg: arg_1 AND arg_2 OR arg_3
+        expectedResults:
+            1. Expression must be created successfully.
+
+    Bugzilla:
+        1506672
+        1660460
+        1718895
+    """
+    filter_name = fauxfactory.gen_alphanumeric()
+
+    if len(running_vms) >= 2:
+        vm1_name, vm2_name = running_vms[0], running_vms[1]
+        vm1 = appliance.collections.infra_vms.instantiate(vm1_name, provider)
+        vm2 = appliance.collections.infra_vms.instantiate(vm2_name, provider)
+    else:
+        pytest.skip("Less than 2 VMs are running, that's not enough for multiple condition test")
+
+    # add the same tags as in the customer case
+    tag1 = (appliance.collections.categories.instantiate(display_name="Environment")
+            .collections.tags.instantiate(display_name="Development"))
+    tag2 = (appliance.collections.categories.instantiate(display_name="Environment")
+            .collections.tags.instantiate(display_name="Test"))
+    vm1.add_tag(tag1)
+    vm2.add_tag(tag2)
+
+    vm_view = navigate_to(appliance.collections.infra_vms, 'VMsOnly')
+    # Find the number of VMs before filtering
+    vms_before = len(vm_view.entities.get_all())
+    # Set up the filter
+    vm_view.entities.search.advanced_search(
+        "fill_field(Virtual Machine : Power State, INCLUDES, on);select_first_expression;click_and;"
+        "fill_tag(Virtual Machine.My Company Tags : Environment, Development);"
+        "select_last_expression; click_or;"
+        "fill_tag(Virtual Machine.My Company Tags : Environment, Test)", filter_name)
+
+    @request.addfinalizer
+    def cleanup():
+        if not BZ(1725838, forced_streams=["5.10", "5.11"]).blocks:
+            # the screen after applied search is unclickable
+            vm_view.entities.search.load_filter(filter_name)
+            vm_view.entities.search.delete_filter()
+            vm1.remove_tag(tag1)
+            vm2.remove_tag(tag2)
+
+    # Check the number of VMs after filtering and that it's less than without filtering
+    vms_after = len(vm_view.entities.get_all())
+
+    assert vms_after < vms_before
```

Comment:
Are you not directly asserting that `vms_after == 2` because there may be other vms with these tags already?

If so, consider asserting that `2 <= vms_after < vms_before`

Reply:
yeah, there's a low probability there're other VMs with such tags but I'd prefer to stay on the safe side and go with the second options